app/src/main/kotlin/com/hippo/ehviewer/ui/legacy/FabLayout.kt

```diff
@@ -1,482 +1,134 @@
-                } else {
-                    isClickable = false
+    @Composable
+    override fun Content() {
+        Mdc3Theme {
+            if (expanded) {
+                DisposableEffect(Unit) {
+                    listeners.forEach { it.invoke(true) }
+                    onDispose {
+                        listeners.forEach { it.invoke(false) }
+                    }
                 }
-            }
-        }
-    }
-
-    fun toggle() {
-        isExpanded = !mExpanded
-    }
-
-    var isExpanded: Boolean
-        get() = mExpanded
-        set(expanded) {
-            setExpanded(expanded, true)
-        }
-
-    fun setExpanded(expanded: Boolean, animation: Boolean) {
-        if (mExpanded != expanded) {
-            mExpanded = expanded
-            if (mAutoCancel) {
-                if (expanded) {
-                    setOnClickListener(this)
-                } else {
-                    isClickable = false
+                if (autoCancel) {
+                    Spacer(
+                        modifier = Modifier.fillMaxSize().pointerInput(expanded) {
+                            detectTapGestures {
+                                expanded = false
+                            }
+                        },
+                    )
                 }
             }
-            val count = childCount
-            if (count > 0) {
-                if (mMainFabCenterY == -1f || !animation) {
-                    // It is before first onLayout
-                    val checkCount = if (mHidePrimaryFab) count else count - 1
-                    for (i in 0 until checkCount) {
-                        val child = getChildAt(i)
-                        if (child.visibility == GONE) {
-                            continue
-                        }
-                        child.visibility = if (expanded) VISIBLE else INVISIBLE
-                        if (expanded) {
-                            child.alpha = 1f
-                        }
-                    }
-                } else {
-                    if (mHidePrimaryFab) {
-                        setPrimaryFabAnimation(getChildAt(count - 1), expanded, !expanded)
+            val coroutineScope = rememberCoroutineScope()
+            val appearState by animateFloatAsState(
+                targetValue = if (hidden) 0f else 1f,
+                animationSpec = tween(300),
+                label = "hiddenState",
+            )
+            Box(
+                modifier = Modifier.fillMaxSize().padding(16.dp),
+                contentAlignment = Alignment.BottomEnd,
+            ) {
+                Box(contentAlignment = Alignment.Center) {
+                    val animatedProgress by animateFloatMergePredictiveBackAsState(expandedBackingField)
+                    FloatingActionButton(
+                        onClick = { expanded = !expanded },
+                        modifier = Modifier.rotate(lerp(90f, 0f, appearState)).scale(appearState),
```

Comment:
I believe it was counterclockwise for hiding before?

Reply:
Done